### PR TITLE
Add command to install cert generated by openssl extension

### LINKED ISSUE #17: Use HTTPS by default
HTTPS should be used by default when setting up a local site using `composer chassis init`.

If it's not possible to do this by default, a command that looks something like `composer chassis secure` would be handy. [Laravel Valet provides this](https://laravel.com/docs/5.8/valet#securing-sites) and it's pretty neat.

### PR DESCRIPTION
**Pull request by roborourke**
Blocked by 
- [x] https://github.com/humanmade/platform-chassis-extension/pull/4
- [x] https://github.com/Chassis/chassis-openssl/pull/2

Fixes #17

This needs some testing still, in particular on Windows.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -44,3 +45,26 @@ We recommend the following common development tools:
...
+
+You should now be able to browse your local environment via HTTPS without certificate warnings.
+
+**Note:** this command only supports OSX and Windows currently.
+
+### Firefox
+
+Because the Firefox browser uses its own certificate store you will either need to install the generated certificate file manually or alternatively follow these steps:
+
+1. Open Firefix
```
Comment (johnbillion) [inline docs/README.md]: Typo

File: docs/README.md [hunk 2]
```diff
@@ -44,3 +45,26 @@ We recommend the following common development tools:
 * [Mailhog](https://github.com/Chassis/MailHog) - Captures outbound email from Altis and provides a fake inbox
 
 Consult the [Chassis documentation](http://docs.chassis.io/en/latest/extend/) for information about installing additional extensions.
+
+
+## Using HTTPS locally
+
+Local Chassis will generate an SSL certificate you can use to run your local environment over HTTPS. The file will be located in the `/chassis` directory, by default it will be called `altis.local.crt` but if you have customised the `hosts` in `config.local.yaml` it will use the first host name in that list for the file name.
```
Comment (johnbillion) [inline docs/README.md]: Is the extension for the cert file `.crt` or `.cert`? Here it's `.crt` but in `$cert_file` in the `Command::secure()` method it's `.cert`.

File: inc/class-command.php [hunk 3]
```diff
@@ -139,7 +142,17 @@ protected function init( InputInterface $input, OutputInterface $output ) {
 			return;
 		}
 
-		return $this->run_command( 'vagrant up' );
+		$this->start( $input, $output );
+
+		// And run the initial setup, if the user wants to.
+		$question = new ConfirmationQuestion( 'Install SSL certificate? [Y/n] ', true );
```
Comment (johnbillion) [inline inc/class-command.php]: Minor: Nothing uses `SSL` any more so this should say `security certificate` or maybe `HTTPS certificate`.

File: inc/class-command.php [hunk 4]
```diff
@@ -182,4 +195,43 @@ protected function status( InputInterface $input, OutputInterface $output ) {
...
+			return 1;
+		}
+
+		// Fimd cert file.
+		$config = Yaml::parseFile( $config_file );
+		$cert_file = $config['hosts'][0] . '.cert';
+		$cert_path = $chassis_dir . DIRECTORY_SEPARATOR . $cert_file;
+
+		if ( ! file_exists( $cert_path ) ) {
+			$output->writeln( sprintf( '<warning>The SSL certificate file "%s" does not exist yet. Run `composer chassis start` first to provision the VM and generate the file.</warning>', $cert_file ) );
```
Comment (johnbillion) [inline inc/class-command.php]: Minor: Reference to `SSL` here too.

File: inc/class-command.php (hunk 3 above)
Comment (johnbillion) [inline inc/class-command.php]: Minor: Nothing uses `SSL` any more so this should say `security certificate` or maybe `HTTPS certificate`.

**Comment by roborourke** [comment]
Cheers @johnbillion, settled on HTTPS certificate as it feels more explicit. Thinking we could add a config option to force HTTPS too.

**Comment by roborourke** [comment]
@johnbillion ready for a final review now

File: inc/class-command.php [hunk 5]
```diff
@@ -65,6 +65,15 @@ protected function execute( InputInterface $input, OutputInterface $output ) {
 			case 'stop':
 				return $this->stop( $input, $output );
 
+			case 'provision':
+				return $this->provision( $input, $output );
+
+			case 'ssh':
```
Comment (joehoyle) [inline inc/class-command.php]: Let's rename this to `shell`, to match `local-server`.

Comment (roborourke) [reply inc/class-command.php]: Moved to another PR to make the changelog cleaner.

File: docs/README.md [hunk 6]
```diff
@@ -44,3 +45,26 @@ We recommend the following common development tools:
 * [Mailhog](https://github.com/Chassis/MailHog) - Captures outbound email from Altis and provides a fake inbox
 
 Consult the [Chassis documentation](http://docs.chassis.io/en/latest/extend/) for information about installing additional extensions.
+
+
+## Using HTTPS locally
+
+Local Chassis will generate an HTTPS security certificate you can use to run your local environment over HTTPS. The file will be located in the `/chassis` directory, by default it will be called `altis.local.cert` but if you have customised the `hosts` in `config.local.yaml` it will use the first host name in that list for the file name.
```
Comment (joehoyle) [inline docs/README.md]: Just wanted to make sure that this cert actually will work for all the hostsnames you add to your config? Currently we have sites that use a chassis config with subdomains too.

Comment (roborourke) [reply docs/README.md]: Goood question. It did but looking at this now https://github.com/Chassis/chassis-openssl/blob/master/modules/openssl-nginx/templates/cert.cnf.erb#L41

Seems like that might be a bug. 🕷

Comment (roborourke) [reply docs/README.md]: So not a bug, it references the `[ alt_names ]` section of the cnf file, however we weren't generating DNS entries for all items under the hosts key, fixed in https://github.com/Chassis/chassis-openssl/pull/3

**Pull request merged**